test/parallel/test-tls-multi-key.js

```diff
@@ -2,52 +2,162 @@
+  passphrase: 'sample',
+  ca: [
+    fs.readFileSync(common.fixturesDir + '/keys/ec-cert.pem'),
+    fs.readFileSync(common.fixturesDir + '/keys/ca1-cert.pem'),
+  ],
+});
+
+// Key and cert with mixed algorithms, and cert chains with intermediate CAs
+test({
+  key: [
+    fs.readFileSync(common.fixturesDir + '/keys/ec10-key.pem'),
+    fs.readFileSync(common.fixturesDir + '/keys/agent10-key.pem'),
+  ],
+  cert: [
+    fs.readFileSync(common.fixturesDir + '/keys/agent10-cert.pem'),
+    fs.readFileSync(common.fixturesDir + '/keys/ec10-cert.pem'),
+  ],
+  rsaCN: 'agent10.example.com',
+  eccCN: 'agent10.example.com',
+  ca: [
+    fs.readFileSync(common.fixturesDir + '/keys/ca2-cert.pem'),
+    fs.readFileSync(common.fixturesDir + '/keys/ca5-cert.pem'),
+  ],
+});
+
+// Key and cert with mixed algorithms, and cert chains with intermediate CAs,
+// using PFX for EC.
+test({
+  key: [
+    fs.readFileSync(common.fixturesDir + '/keys/agent10-key.pem'),
+  ],
+  cert: [
+    fs.readFileSync(common.fixturesDir + '/keys/agent10-cert.pem'),
+  ],
+  pfx: fs.readFileSync(common.fixturesDir + '/keys/ec10.pfx'),
+  passphrase: 'sample',
+  rsaCN: 'agent10.example.com',
+  eccCN: 'agent10.example.com',
+  ca: [
+    fs.readFileSync(common.fixturesDir + '/keys/ca2-cert.pem'),
+    fs.readFileSync(common.fixturesDir + '/keys/ca5-cert.pem'),
+  ],
+});
+
+// Key and cert with mixed algorithms, and cert chains with intermediate CAs,
+// using PFX for RSA.
+test({
+  key: [
+    fs.readFileSync(common.fixturesDir + '/keys/ec10-key.pem'),
+  ],
+  cert: [
+    fs.readFileSync(common.fixturesDir + '/keys/ec10-cert.pem'),
+  ],
+  pfx: fs.readFileSync(common.fixturesDir + '/keys/agent10.pfx'),
+  passphrase: 'sample',
+  rsaCN: 'agent10.example.com',
+  eccCN: 'agent10.example.com',
+  ca: [
+    fs.readFileSync(common.fixturesDir + '/keys/ca2-cert.pem'),
+    fs.readFileSync(common.fixturesDir + '/keys/ca5-cert.pem'),
+  ],
+});
+
+function test(options) {
+  const rsaCN = options.rsaCN || 'agent1';
+  const eccCN = options.eccCN || 'agent2';
+  const ca = options.ca;
+  const server = tls.createServer(options, function(conn) {
+    conn.end('ok');
+  }).listen(0, common.mustCall(connectWithEcdsa));
+
+  function connectWithEcdsa() {
+    const ecdsa = tls.connect(this.address().port, {
+      ciphers: 'ECDHE-ECDSA-AES256-GCM-SHA384',
+      rejectUnauthorized: true,
+      ca: ca,
+      checkServerIdentity: (_, c) => assert.strictEqual(c.subject.CN, eccCN),
+    }, common.mustCall(function() {
+      assert.deepStrictEqual(ecdsa.getCipher(), {
+        name: 'ECDHE-ECDSA-AES256-GCM-SHA384',
+        version: 'TLSv1/SSLv3'
+      });
+      assert.strictEqual(ecdsa.getPeerCertificate().subject.CN, eccCN);
+      // XXX(sam) certs don't currently include EC key info, so depend on
+      // absence of RSA key info to indicate key is EC.
+      assert(!ecdsa.getPeerCertificate().exponent, 'not cert for an RSA key');
```

Comment:
We do not worry about to check peerCert of ECC attribute. It is enough to confirm that ECDSA cipher has been established if ECC cert is surely working.  But this check does not have harm.